packages/pricing/src/models/price-list-rule.ts

```diff
@@ -1,56 +1,92 @@
+import { DAL } from "@medusajs/types"
+import {
+  DALUtils,
+  createPsqlIndexStatementHelper,
+  generateEntityId,
+} from "@medusajs/utils"
 import {
   BeforeCreate,
   Cascade,
   Collection,
   Entity,
+  Filter,
   ManyToOne,
-  OneToMany,
   OnInit,
+  OneToMany,
   OptionalProps,
   PrimaryKey,
-  Unique,
+  Property,
 } from "@mikro-orm/core"
-
-import { generateEntityId } from "@medusajs/utils"
 import PriceList from "./price-list"
 import PriceListRuleValue from "./price-list-rule-value"
 import RuleType from "./rule-type"
 
-type OptionalFields = "id"
-type OptionalRelations = "rule_type" | "price_list"
+type OptionalFields = DAL.SoftDeletableEntityDateColumns
+
+const tableName = "price_list_rule"
+const PriceListRuleDeletedAtIndex = createPsqlIndexStatementHelper({
+  tableName: tableName,
+  columns: "deleted_at",
+  where: "deleted_at IS NOT NULL",
+})
+
+const PriceListRuleRuleTypeIdIndex = createPsqlIndexStatementHelper({
+  tableName: tableName,
+  columns: "rule_type_id",
+  where: "deleted_at IS NOT NULL",
+  unique: true,
+})
 
-@Entity()
-@Unique({
-  name: "IDX_price_list_rule_rule_type_id_price_list_id_unique",
-  properties: ["price_list", "rule_type"],
+const PriceListRulePriceListIdIndex = createPsqlIndexStatementHelper({
+  tableName: tableName,
+  columns: "price_list_id",
+  where: "deleted_at IS NOT NULL",
 })
+
+@Entity({ tableName })
+@Filter(DALUtils.mikroOrmSoftDeletableFilterOptions)
 export default class PriceListRule {
-  [OptionalProps]: OptionalFields | OptionalRelations
+  [OptionalProps]: OptionalFields
 
   @PrimaryKey({ columnType: "text" })
   id!: string
 
-  @ManyToOne({
-    entity: () => RuleType,
-    fieldName: "rule_type_id",
-    name: "price_rule_rule_type_id_unique",
-    index: "IDX_price_list_rule_rule_type_id",
-  })
+  @PriceListRuleRuleTypeIdIndex.MikroORMIndex()
+  @ManyToOne({ entity: () => RuleType, fieldName: "rule_type_id" })
   rule_type: RuleType
```

Comment:
todo:

```ts
@PriceListRuleRuleTypeIdIndex.MikroORMIndex()
@ManyToOne(() => RuleType, {
    columnType: "text",
    onDelete: "cascade",
    fieldName: "rule_type_id",
    mapToPk: true
})
rule_type_id: string

@ManyToOne(() => RuleType, {
  persist: false,
})
rule_type: RuleType

/* ... */

  @BeforeCreate()
  beforeCreate() {
    this.id = generateEntityId(this.id, "plrule")
    this.rule_type_id ??= this.rule_type?.id
  }
  @OnInit()
  onInit() {
    this.id = generateEntityId(this.id, "plrule")
    this.rule_type_id ??= this.rule_type?.id
  }
```

Reply:
now you don't necessarely have to provide a rule_type object to assign it, you can just provide the rule_type_id, and if you still provide a rule_type, it will grab the id automatically

Reply:
You can use that everywhere you have a many to one to start with

Reply:
This will be changed in the follow up iterations. Its already on my todo. Would like to keep this PR focused.

Reply:
sounds good to me :D